src/components/shoppingCart/cartdetail/CartDetail.jsx

```diff
@@ -3,11 +3,54 @@ import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
 import Button from "react-bootstrap/Button";
-import { MOCK_DATABASE } from "../../common/MockDatabase";
+import "./CartDetail.scss";
+import {
+  deleteItemFromCartAction,
+  increaseProductQuantity,
+  decreaseProductQuantity,
+  sumTotalPrice,
+  addOneMoreItem,
+  removeOneMoreItem,
+} from "../../redux/actions";
+import { useDispatch, useSelector } from "react-redux";
 
-const { title, images, brand, price } = MOCK_DATABASE[0];
+const CartDetail = (props) => {
+  const dispatch = useDispatch();
+  const productsData = useSelector((state) => state.addOrDeleteProductData);
+  // Getting all the total prices from individual items and add them together
+  const itemsTotalPricesArray = useSelector((state) => state.getTotalPrice);
+  const allItemsTotalPrice = itemsTotalPricesArray.reduce((cur, acc) => {
+    return cur + acc;
+  }, 0);
+
+  // Get total price from one item
+  React.useEffect(() => {
+    dispatch(sumTotalPrice(props.info.price, props.info.quantity));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
are you sure you only want this to run once? this seems like it would be a bug to me, if the props change you would want to recompute the total price

Reply:
When we increase or decrease the quantity in the shopping cart this component  rerenders  automatically  (line 125 , 126 , 135,136)